### DISCUSSION TITLE
Proxy on '/' as exact path, not '/*'

### DISCUSSION THREAD

**Discussion by bleszerd** [chimurai/http-proxy-middleware#767] [Q&A] [answered]
Hey guys.

How do I set a rule for a `"/users"` route that doesn't apply to `"/users/find"`?

I need in my project to distinguish a public route from a private one, but the `"/users"` rules propagate to all endpoints starting with the `"/users"` prefix, like `"users/find"` or `"users/updateName"`

**Comment by chimurai** [comment]
For v2:
You can configure a glob pattern or write your own function:
https://github.com/chimurai/http-proxy-middleware/blob/2.x/recipes/context-matching.md#wildcard--exclusion

For v3@beta:
Same functionality has been moved to `pathFilter`:
https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/pathFilter.md#path-filter

**Reply by bleszerd** [reply]
Thanks, I think this solves my problem.

I even believe that making this explanation more accessible is a good idea, I recommend putting in the project's README.me a reference to the link you sent me

**Reply by chimurai** [reply]
Believe it's already documented in the README.md.

Open for suggestions to improve it. 👌

**Comment by bleszerd** [comment, marked as answer]
just updating if they find this question: you can use `^/user$` as exact path and `^/user/**$` to match all routes that have `/user` as prefix.

You might want to do something like:

**[Typescript]**
```ts
const createRoute = ({
  redirectHost,
  route,
  useAsPrefix = true,
}: IRouteOptions) => {
  const routePath = useAsPrefix ? `${route}/**` : route;

  router.use(
    `^/oldApi/${routePath}$`,
    createProxyMiddleware({
      pathRewrite: {
        '^/oldApi': '/newApi',
      },
      target: redirectHost,
    }),
  );
};
```

@chimurai thanks for the help
